tests/confidence/healthcheck.sh

```diff
@@ -0,0 +1,55 @@
+#################################################################################################
+###
+### CURL the deployed containers healthcheck and return the status and SHA.
+### check the SHA against a passed in parameter
+###
+### Input parameters ( not validated )
+### 1 URL
+### 2 SHA
+###
+### Returns
+### 1 on failure
+### 0 on sucess
+###
+#################################################################################################
+URL=${1}
+EXPECTED_SHA=${2}
+
+#URL="get-into-teaching-api-dev"
+#EXPECTED_SHA="730ff0a"
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m' 
+
+rval=0
+FULL_URL="https://${URL}.london.cloudapps.digital/api/operations/health_check"
+http_status=$(curl -o /dev/null -s -w "%{http_code}"  ${FULL_URL}) 
+if [ "${http_status}" != "200" ] 
+then
+	echo "${RED}HTTP Status ${http_status}${NC}"
+	rval=1
+else
+	echo "${GREEN}HTTP Status is Healthy${NC}"
+
+        json=$(curl -s -X GET ${FULL_URL})
+        status=$( echo ${json} | jq -r .status)
+        if [ "${status}" != "healthy" ] 
```

Comment:
This can also be 'degraded' as well as 'unhealthy', not sure if you want to differentiate here or not though

Reply:
I really can only handle a true / false state so if it is not healthy, then fail it, we can investigate. It is only being run after deployment and I would hope it was ok then

Reply:
It may be worth checking for `unhealthy` explicitly in that case - we probably don't want to block the pipeline if it's degraded as that indicates a downstream service is having issues

Reply:
OK since I need to remove colour I will